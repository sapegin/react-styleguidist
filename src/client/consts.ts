export const ExampleModes = Object.freeze({
	hide: 'hide',
	collapse: 'collapse',
	expand: 'expand',
});

export const UsageModes = Object.freeze({
	hide: 'hide',
	collapse: 'collapse',
	expand: 'expand',
});
