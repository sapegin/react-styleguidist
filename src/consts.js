/* eslint-disable import/prefer-default-export */

export const DisplayModes = Object.freeze({
	// Show all sections and components (default)
	all: 'all',
	// Show one section
	section: 'section',
	// Show one component
	component: 'component',
	// Show one example inside component or section
	example: 'example',
	// TODO: error (404)
});

export const ExampleMode = Object.freeze({
	hide: 'hide',
	collapse: 'collapse',
	expand: 'expand',
});

export const UsageMode = Object.freeze({
	hide: 'hide',
	collapse: 'collapse',
	expand: 'expand',
});
