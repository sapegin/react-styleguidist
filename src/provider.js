import React, { Component } from 'react';
import PropTypes from 'prop-types';

export const Context = React.createContext();

export function Consumer(WrappedComponent) {
	const componentName = WrappedComponent.name.replace(/Renderer$/, '');
	return class extends Component {
		static displayName = `Consumer(${componentName})`;
		state = {};
		render() {
			return (
				<Context.Consumer>
					{context => {
						if (!context) {
							throw new Error(`Consumer cannot be rendered outside the Provider component`);
						}
						return <WrappedComponent {...context} {...this.props} />;
					}}
				</Context.Consumer>
			);
		}
	};
}

export function Provider(props) {
	const { children, ...rest } = props;
	const ui = typeof children === 'function' ? children(this.state) : children;

	return <Context.Provider value={{ ...rest }}>{ui}</Context.Provider>;
}

Provider.propTypes = {
	children: PropTypes.any,
	codeRevision: PropTypes.number.isRequired,
	config: PropTypes.object.isRequired,
	slots: PropTypes.object.isRequired,
	displayMode: PropTypes.string,
};
