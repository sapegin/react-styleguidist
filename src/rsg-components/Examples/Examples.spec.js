import React from 'react';
import noop from 'lodash/noop';
import Examples from '../Examples';
import { DisplayModes } from '../../consts';
import { Provider } from '../../provider';

const examples = [
	{
		type: 'code',
		content: '<button>OK</button>',
		evalInContext: noop,
	},
	{
		type: 'markdown',
		content: 'Hello *world*!',
	},
];

it('should render examples', () => {
	const context = {
		codeRevision: 1,
		displayMode: DisplayModes.example,
	};
	const actual = shallow(
		<Provider {...context}>
			<Examples examples={examples} name="button" exampleMode="collapse" />
		</Provider>
	);

	expect(actual).toMatchSnapshot();
});

it('should not render a example with unknown type', () => {
	const faultyExample = [
		{
			type: 'unknown',
			content: 'FooBar',
		},
	];
	const context = {
		codeRevision: 1,
	};

	const actual = mount(
		<Provider {...context}>
			<Examples examples={faultyExample} name="button" exampleMode="collapse" />
		</Provider>
	);
	const article = actual.find('article');
	expect(article.length).toEqual(1);
	expect(article.text().includes('FooBar')).toEqual(false);
});
