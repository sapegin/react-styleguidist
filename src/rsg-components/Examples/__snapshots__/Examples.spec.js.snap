// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render examples 1`] = `
<ContextProvider
  value={
    Object {
      "codeRevision": 1,
      "displayMode": "example",
    }
  }
>
  <Consumer(Examples)
    exampleMode="collapse"
    examples={
      Array [
        Object {
          "content": 
<button>
  OK
</button>
,
          "evalInContext": [Function],
          "type": "code",
        },
        Object {
          "content": "Hello *world*!",
          "type": "markdown",
        },
      ]
    }
    name="button"
  />
</ContextProvider>
`;
